utils/clients/tools/tests/test_isisicat_prefix_mapping.py

```diff
@@ -0,0 +1,87 @@
+    @patch('utils.clients.tools.isisicat_prefix_mapping.AUTOREDUCTION_INSTRUMENT_NAMES',
+           ['ENGINGX'])
+    @patch('utils.clients.icat_client.ICATClient.execute_query')
+    def test_fetch_instrument_fullname_mapping_executes_icat_query(self, mock_execute_query, _):
+        """
+        Test: If instruments are fetched through a query
+        When: Testing if fetch_instrument_fullname_mapping executes an ICAT query
+        """
+        fetch_instrument_fullname_mapping()
+        mock_execute_query.assert_called_once()
+
+    @patch('icat.Client.__init__', return_value=None)
+    @patch('utils.clients.tools.isisicat_prefix_mapping.AUTOREDUCTION_INSTRUMENT_NAMES',
+           ['INVALIDINSTUMENT'])
+    @patch('logging.Logger.warning')
+    @patch('utils.clients.icat_client.ICATClient.execute_query',
+           returns_value=[MockInstrumentQueryResult("ABCDE", "ABC")])
+    def test_fetch_instrument_fullname_mapping_log_invalid_instrument(self, _mock_execute_query,
+                                                                      mock_logger_warning, _):
+        """
+        Test: If invalid instrument name in utils.settings.VALID_INSTRUMENTS is logged as not found
+        When: Testing if fetch_instrument_fullname_mapping picks up invalid instruments
+        """
+        fetch_instrument_fullname_mapping()
+        mock_logger_warning.assert_called_once()
+
+    @patch('icat.Client.__init__', return_value=None)
+    @patch("utils.clients.tools.isisicat_prefix_mapping.AUTOREDUCTION_INSTRUMENT_NAMES",
+           ["ENGINX", "GEM"])
+    @patch('utils.clients.icat_client.ICATClient.execute_query',
+           return_value=[MockInstrumentQueryResult("ENG", "ENGINX"),
+                         MockInstrumentQueryResult("GEM", "GEM")])
+    def test_icat_prefix_mapping_length(self, _mock_instruments, _mock_execute_query):
+        """
+        Test: fetch_instrument_fullname_mapping produces the same number of results as stored in
+              utils.settings.VALID_INSTRUMENTS
+        When: Called when testing to see if the correct number of instruments in prefix mapping
+        """
+        prefix_map = fetch_instrument_fullname_mapping()
+        self.assertEqual(2, len(prefix_map.keys()))
+
+    @patch('icat.Client.__init__', return_value=None)
+    @patch('utils.clients.icat_client.ICATClient.execute_query')
+    @patch("utils.clients.tools.isisicat_prefix_mapping.AUTOREDUCTION_INSTRUMENT_NAMES", ["ENGINX"])
+    def test_icat_prefix_mapping(self, mock_execute_query, _):
+        """
+        Test: If fetch_instrument_fullname_mapping properly maps instrument names map to ICAT
+              instrument prefixes using utils.settings.VALID_INSTRUMENTS
+        When: Called when testing correct mapping
+        """
+        icat_test_instrument = ("ENG", "ENGINX")
+        mock_execute_query.return_value = [MockInstrumentQueryResult(*icat_test_instrument)]
+
+        prefix_map = fetch_instrument_fullname_mapping()
```

Comment:
NITPICK: could we re-word this to "actual"

Reply:
Which line?

Reply:
Change prefix_map to `actual = fetch_instrument_fullname_mapping()`